lib/charms/mongodb/v1/mongos.py

```diff
@@ -286,3 +302,84 @@ def _log_removal_info(self, removal_info, shard_name):
             str(remaining_chunks),
             ",".join(dbs_to_move),
         )
+
+    def _retrieve_remaining_chunks(self, removal_info) -> int:
+        """Parses the remaining chunks to remove from removeShard command."""
+        return removal_info["remaining"]["chunks"] if "remaining" in removal_info else 0
+
+    def _move_primary(self, databases_to_move: List[str], old_primary: str) -> None:
+        """Moves all the provided databases to a new primary.
+
+        Raises:
+            NotEnoughSpaceError, ConfigurationError, OperationFailure
+        """
+        for database_name in databases_to_move:
+            db_size = self.get_db_size(database_name, old_primary)
+            new_shard, avail_space = self.get_shard_with_most_available_space(
+                shard_to_ignore=old_primary
+            )
+            if db_size > avail_space:
+                no_space_on_new_primary = (
+                    f"Cannot move primary for database: {database_name}, new shard: {new_shard}",
+                    f"does not have enough space. {db_size} > {avail_space}",
+                )
+                logger.error(no_space_on_new_primary)
+                raise NotEnoughSpaceError(no_space_on_new_primary)
+
+            # From MongoDB Docs: After starting movePrimary, do not perform any read or write
+            # operations against any unsharded collection in that database until the command
+            # completes.
+            logger.info(
+                "Moving primary on %s database to new primary: %s. Do NOT write to %s database.",
+                database_name,
+                new_shard,
+                database_name,
+            )
+            # This command does not return until MongoDB completes moving all data. This can take
+            # a long time.
+            self.client.admin.command("movePrimary", database_name, to=new_shard)
```

Comment:
Just curious: Why won't we put charm in a Maintenance status here, if the operation can take a long time?

Reply:
`mongos.py` has no reference to `self.charm` however the function that calls this goes into maintenance status:
[config server side](https://github.com/canonical/mongodb-operator/blob/7837e28be6f615e8017e24518542012f1c6405a8/lib/charms/mongodb/v1/shards_interface.py#L243)

And this is similarly reported on: 
[shard side](https://github.com/canonical/mongodb-operator/blob/7837e28be6f615e8017e24518542012f1c6405a8/lib/charms/mongodb/v1/shards_interface.py#L412)